### ISSUE TITLE
Screen tearing at the top on intel hd620

### ISSUE THREAD

**Issue by montvid** [compiz-reloaded/compiz#122]
I have Screen tearing at the top of the screen on intel hd620 using fedora 28 mate+compiz. There is no screen tearing while browsing with "sync to vblank" enabled but there is screen tearing at the top of the screen while using modesetting and intel driver. Only intel driver with dri2 fixes screen tearing at the top as the modesetting driver does not at all. So DRI3 is not working with mate+compiz at all in intel & modesetting driver.
See here it seems mate only recently got dri3 support maybe it has something to do with compiz too? 
https://github.com/mate-desktop/marco/issues/326

Does compiz have tear free dri3 support on mate? How can this problem be fixed?

**Comment by montvid** [comment]
I mean there is tearing at the middle that sync to vblank ON fixes and there is tearing at the top that only intel drivers downgraded to dri2 fix.

**Comment by alexarnaud** [comment]
Hello montvid,

The bug you've is not related to Compiz 0.9 but to Compiz reloaded 0.8 
on Fedora. It's a fork of Compiz so to report the issue go to 
https://github.com/compiz-reloaded

Could you close this issue so?

Best regards,
Alex.

Le 02/05/2018 à 00:17, montvid a écrit :

**Comment by montvid** [comment]
Am I lost or are you lost? Your link is generic so does not point to a repo. I see 
https://github.com/compiz-reloaded/compiz is
"Compiz 0.8.x core project" so I am posting to the right repo.

**Comment by alexarnaud** [comment]
You're right, I'm completely lost, I've misreading the origin of the bug, sorry

Best regards,
Alex.

**Comment by soreau** [comment]
In your first post you said 

> There is no screen tearing while browsing with "sync to vblank" enabled

By this do you mean 'Sync to VBlank' option in ccsm eliminates tearing regardless of which driver or dri version used? Because that would be the expected behavior.

By 'tearing at the top' do you mean when scrolling? rotating cube? something else?

**Comment by montvid** [comment]
Yes, 'Sync to VBlank' seems to work as expected but the tearing at the top is happening regardless of it being on or off. By tearing at the top i mean I see it by scrolling in firefox fullscreen mode but most importantly by dragging a window horizontally like from left to right. I tried to record the tearing but the recording shows that there is no tearing :D DRI2 fixes the tearing but maybe compiz can be fixed to work tearfree with dri3 too?

**Comment by montvid** [comment]
It seems there is no screen tearing when I maximise or minimize a window, only when I move it horizontally... :/

**Comment by soreau** [comment]
You might try DRI3 with Option "TearFree" 1

**Comment by montvid** [comment]
I tried that before. I told you it is with intel radeon and modesetting drivers but only in compiz. I made a video with my phone see here as I dont think you understand me: https://drive.google.com/file/d/1aeqo1_LaLuUDWZV-BOue5Wl-Tk3bjc8b/

**Comment by montvid** [comment]
@soreau what video driver configuration do you use to run compiz tearfree?

**Comment by soreau** [comment]
I use amdgpu with tearfree option. Maybe try upgrading mesa and possibly kernel.
